### PR TITLE
csvcubed workflows minus pandas, gssutils & docs

### PR DESCRIPTION
**Pull request by CharlesRendle**
PR to add a similar github workflow structure to that of csvcubed when making pushing changes to a PR, merging to main and then releasing. Addressing ticket #16 

I have removed the sections and references to behave testing as there are currently no behave tests in this repo.

I have also updated the dockerfile so that we can test the code locally using python 3.11.1 which exposed a failing unit test. This was introduced by an update to `datetime.datetime.fromisoformat` as this was updated to parse most ISO 8601 formats. Previously we expected to raise a value error. Now we check which python version is in use and assert accordingly.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,15 +8,20 @@ jobs:
     strategy:
       matrix:
         # Test in all supported versions of python in both Windows and Ubuntu.
-        python-version: ['3.11']
-        os: [ubuntu-latest]
+        python-version: ['3.9', '3.10', '3.11']
```
Comment (canwaf) [inline .github/workflows/release.yaml:11]: Are we still testing 3.10? I thought we were keeping 3.9 and 3.11?

**Review by canwaf — COMMENTED** [review]
Just one comment.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (CharlesRendle) [reply .github/workflows/release.yaml:11]: On release, we are testing all the python versions for which we support. Same as with the csvcubed release.

Comment (CharlesRendle) [reply .github/workflows/release.yaml:11]: All python *minor* versions

**Review by canwaf — APPROVED** [review]
Approved as discussed.

**Pull request merged**